datumaro/plugins/data_formats/segment_anything/exporter.py

```diff
@@ -0,0 +1,154 @@
+
+from pycocotools import mask as mask_utils
+
+from datumaro.components.annotation import AnnotationType
+from datumaro.components.errors import DatumaroError, MediaTypeError
+from datumaro.components.exporter import Exporter
+from datumaro.components.media import Image
+from datumaro.util import NOTSET, dump_json_file
+
+
+def replace(json_data, key, value_new):
+    value_origin = json_data[key]
+    if value_origin is NOTSET:
+        json_data[key] = value_new
+        return
+    if value_origin != value_new:
+        raise DatumaroError(f"The value for '{key}' is not same for item {value_new}")
+
+
+class SegmentAnythingExporter(Exporter):
+    DEFAULT_IMAGE_EXT = ".jpg"
+
+    def __init__(
+        self,
+        extractor,
+        save_dir,
+        **kwargs,
+    ):
+        super().__init__(extractor, save_dir, **kwargs)
+
+    def apply(self):
+        if self._extractor.media_type() and not issubclass(self._extractor.media_type(), Image):
+            raise MediaTypeError("Media type is not an image")
+
+        os.makedirs(self._save_dir, exist_ok=True)
+
+        subsets = self._extractor.subsets()
+        pbars = self._ctx.progress_reporter.split(len(subsets))
+
+        _image_ids = set()
+
+        image_id = 1
+        for pbar, (subset_name, subset) in zip(pbars, subsets.items()):
+            for item in pbar.iter(subset, desc=f"Exporting {subset_name}"):
+                try:
+                    # make sure file_name is flat
+                    file_name = self._make_image_filename(item).replace("/", "__")
+                    image_id = int(item.attributes.get("id", image_id))
+                    while image_id in _image_ids:
+                        image_id += 1
+                    _image_ids.add(image_id)
+
+                    height, width = item.media.size
+                    json_data = {
+                        "image": {
+                            "image_id": image_id,
+                            "file_name": file_name,
+                            "height": height,
+                            "width": width,
+                        },
+                        "annotations": [],
+                    }
+
+                    annotations_grouped = defaultdict(lambda: [])
+                    for annotation in item.annotations:
+                        annotations_grouped[annotation.id].append(annotation)
+
+                    for id, annotations in annotations_grouped.items():
+                        annotation_data = {
+                            "id": id,
+                            "segmentation": NOTSET,
+                            "bbox": NOTSET,
+                            "area": NOTSET,
+                            "predicted_iou": NOTSET,
+                            "stability_score": NOTSET,
+                            "crop_box": NOTSET,
+                            "point_coords": NOTSET,
+                        }
+
+                        for annotation in annotations:
```

Comment:
This loop seems to repeatedly replace the contents of `annotation_data` with each group item. We need to merge the grouped items as same as
https://github.com/openvinotoolkit/datumaro/blob/1c150936f2eb14189b1f48263580ac3b24f58f61/datumaro/plugins/data_formats/coco/exporter.py#L277-L296

Reply:
As commented above, annotations with the same id should represent a same object. `replace` function would raise an error if the contents in `annotation_data` are not the same when the contents are already assigned.

Reply:
Annotations are merged in https://github.com/openvinotoolkit/datumaro/pull/1005/commits/dc0ac4f50f1f0fc9a989c59d688225ee831b53cd